handlers/service_map.go

```diff
@@ -1,20 +1,55 @@
 package handlers
 
 import (
+	"context"
+	"fmt"
 	"net"
 	"sync"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
 )
 
 type ServiceMap struct {
-	services map[string]*net.IP
-	lock     *sync.RWMutex
+	namespace   string
+	client      *containerd.Client
+	context     context.Context
+	containers  map[string]containerd.Container
+	services    map[string]*net.IP
+	servicesPID map[string]uint32
+	lock        *sync.RWMutex
 }
 
-func NewServiceMap() *ServiceMap {
+func NewServiceMap(c *containerd.Client, ns string) *ServiceMap {
+	ctx := namespaces.WithNamespace(context.Background(), "openfaas-fn")
 
 	return &ServiceMap{
-		services: make(map[string]*net.IP),
-		lock:     &sync.RWMutex{},
+		namespace:   ns,
+		client:      c,
+		context:     ctx,
+		containers:  make(map[string]containerd.Container),
+		services:    make(map[string]*net.IP),
+		servicesPID: make(map[string]uint32),
+		lock:        &sync.RWMutex{},
+	}
+}
+
+func (s *ServiceMap) Update() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	containers, _ := s.client.Containers(s.context)
+	for _, k := range containers {
+		i := k.ID()
+		s.containers[i] = k
```

Comment:
Should we really capture a handle to the containerd object, or have some safe DTO struct to keep the key info we want?